jupyter_console/app.py

```diff
@@ -138,21 +138,27 @@ def handle_sigint(self, *args):
             # raise the KeyboardInterrupt if we aren't waiting for execution,
             # so that the interact loop advances, and prompt is redrawn, etc.
             raise KeyboardInterrupt
-    
-    def initialize(self, argv=None):
-        try:
-            super(ZMQTerminalIPythonApp, self).initialize(argv)
-        except NoStart:
-            pass
 
-    def init_code(self):
-        # no-op in the frontend, code gets run in the backend
-        pass
+    @catch_config_error
+    def initialize(self, argv=None):
+        """Do actions after construct, but before starting the app."""
+        super(ZMQTerminalIPythonApp, self).initialize(argv)
+        # create the shell
+        self.init_shell()
```

Comment:
You can add `if self._dispatching: return` after super(), and it should be happy, again.